v2/internal/frontend/desktop/windows/window.go

```diff
@@ -93,3 +86,76 @@ func NewWindow(parent winc.Controller, appoptions *options.App) *Window {
 func (w *Window) Run() int {
 	return winc.RunMainLoop()
 }
+
+func (w *Window) WndProc(msg uint32, wparam, lparam uintptr) uintptr {
+	if w.frontendOptions.Frameless {
+		switch msg {
+		case w32.WM_ACTIVATE:
+			// If we want to have a frameless window but with border, extend the client area outside of the window. This
+			// Option is not affected by returning 0 in WM_NCCALCSIZE.
+			// As a result we have hidden the titlebar but still have the default border drawn.
+			// See: https://docs.microsoft.com/en-us/windows/win32/api/dwmapi/nf-dwmapi-dwmextendframeintoclientarea#remarks
+			if winoptions := w.frontendOptions.Windows; winoptions != nil && winoptions.EnableFramelessBorder {
+				if err := DwmExtendFrameIntoClientArea(w.Handle(), w32.MARGINS{-1, -1, -1, -1}); err != nil {
+					panic(fmt.Sprintf("DwmExtendFrameIntoClientArea failed: %s", err))
+				}
+			}
+		case w32.WM_NCCALCSIZE:
+			// Disable the standard frame by allowing the client area to take the full
+			// window size.
+			// See: https://docs.microsoft.com/en-us/windows/win32/winmsg/wm-nccalcsize#remarks
+			// This hides the titlebar and also disables the resizing from user interaction because the standard frame is not
+			// shown. We still need the WS_THICKFRAME style to enable resizing from the frontend.
+			if wparam != 0 {
+				style := uint32(w32.GetWindowLong(w.Handle(), w32.GWL_STYLE))
+				if style&w32.WS_MAXIMIZE != 0 {
+					// If the window is maximized it must have the normal resizing frame around, if not
+					// some content will be cut-off
+					x := w32.GetSystemMetrics(w32.SM_CXSIZEFRAME)
+					y := w32.GetSystemMetrics(w32.SM_CYSIZEFRAME)
+
+					x, y = w.scaleWithWindowDPI(x, y)
+
+					test := (*w32.RECT)(unsafe.Pointer(lparam))
+					test.Top += int32(y)
+					test.Bottom -= int32(y)
+					test.Left += int32(x)
+					test.Right -= int32(x)
+				}
+
+				return 0
+			}
+		}
+	}
+	return w.Form.WndProc(msg, wparam, lparam)
+}
+
+// TODO this should be put into the winc if we are happy with this solution.
```

Comment:
I think it should be in winc too. Why don't we start here and then move it out after?

Reply:
Yes we could do that, but I think we should at least expose `ScaleWithWindowDPI`.  leaanthony/winc#13 fixes a bug in the scaling and also exposes it.

Reply:
I was able to remove the `ScaleWithWindowDPI` call here.